src/lib/state/selectors/query.js

```diff
@@ -101,6 +104,25 @@ export const updateQueryFilters = (queryFilter, stateFilters) => {
    * convert back to lists
    */
   return filteredStrStates.map(strState => parse(strState));
+}
+
+export const updateQueryFilters = (queryFilter, stateFilters) => {
+  if (_isEmpty(queryFilter)) return;
+
+  /** If we have one filter as query = ['file_type', 'pdf'] */
+  if (!_isArray(_head(queryFilter))) {
+    return updateFilter(queryFilter, stateFilters);
+  }
+
+  /** If we have an array of filters as query we apply the filters one by one.
+   * e.g. query = [['file_type', 'pdf'], ['file_type', 'txt']]
+   */
+  let tempStateFilters = stateFilters;
+  _forEach(
```

Comment:
can you do simply:
```
return queryFilters.filter(queryFilter => updateFilter(queryFilter. stateFilters))
```
?

Reply:
This was my initial approach but as we call updateFilter our stateFilters change, so our next updateFilter should be applied on the altered state and not the initial one. I can explain in a call if needed.